scripts/monitor/visualization/resource_monitoring.py

```diff
@@ -0,0 +1,596 @@
+        """
+        # pandas_bokeh.output_file(f"avg_disk_across_shards_{task}")
+        id_list = self.get_shard_instance_id_list(task)
+        averages = pd.Series([], dtype="float64")
+        for instance_id in id_list:
+            by_id = self.metrics[self.metrics["metrics_instance_id"] == instance_id]\
+                        .sort_values("metrics_timestamp").loc[:, "metrics_disk_used_gb"]
+            row = self.get_runtime_row_from_instance_id(instance_id)
+            requested_disk = float(row["meta_disk_total_gb"].iloc[0][0])
+            data = list()
+            data.append([(float(e[0]) / requested_disk) * 100 for e in by_id])
+            try:
+                averages = averages.append(pd.Series([sum(data[0]) / len(by_id)]),
+                                           ignore_index=True)  # 3870777233638278011 doesn't exist in metrics table
+            except ZeroDivisionError:
+                print(f"the instance id: {instance_id} in meta_runtime does not exist in metrics")
+        # averages.plot_bokeh(kind="scatter", title=f"Average Disk for {task}", xlabel="Shard #", ylabel="Disk % Usage")
+        # return averages.plot\
+        # .line(title=f"Average Disk for {task}", xlabel="Shard #", ylabel="Disk % Usage").get_figure()
+        return averages
+
+    def avg_all_across_shards(self, task: str):
+        """
+        Creates a dataset of the combined data for disk space, memory, and cpu that can be easily made into bokeh plot
+        :param task: task name
+        :return: dataset
+        """
+        disk = self.avg_disk_across_shards(task)
+        mem = self.avg_mem_across_shards(task)
+        cpu = self.avg_cpu_across_shards(task)
+        input_ = [self.task_input_size(task, i) for i in range(len(disk))]
+        typedisk = ["disk %" for _ in range(len(disk))]
+        typemem = ["mem %" for _ in range(len(mem))]
+        typecpu = ["cpu %" for _ in range(len(cpu))]
+        typeinput = ["input GB" for _ in range(len(disk))]
+        x = [i % len(disk) for i in range(len(disk)*4)]
+        # display(type(disk))
+        # display(type(mem))
+        # display(type(cpu))
+        # display(type(pd.Series(data=input_)))
+        y = disk.append(mem).append(cpu).append(pd.Series(data=input_))
+        cat = typedisk + typemem + typecpu + typeinput
+        combine = zip(y, cat, x)
+        dataset = pd.DataFrame(data=tuple(combine), columns=["Percent Usage", "types", "Shard #"])
+        # pandas_bokeh.output_file(f"avg_all_across_shards_{task}home.html")
+        # scatter = dataset.plot_bokeh.scatter(x="Shard #", y="Percent Usage", line_alpha=0, fill_alpha=.6,
+        #                                       category="types", title=Title(text=f"Combined plot for {task}",
+        #                                       align="center"))
+        return dataset
+
+    def total_mem_usage(self) -> None:
+        """
+        Plots global memory usage in the order it's presented in the metrics table
```

Comment:
Not understanding.
For sharded or unsharded?
What's the order in the metrics table?